### ISSUE TITLE
Can we ignore the additional parameter --scale-loss?

### ISSUE THREAD

**Issue by mitming** [facebookresearch/barlowtwins#23]
hi, thanks for your contribution, I am guess that the additional parameter `scale-loss` cannot affect the gradient of loss, so in practice, we can ignore it, please tell me that i am right...

**Comment by StphTphsn** [comment]
Hi, thank you for this judicious remark. It is true that --scale-loss should not affect the gradient of the parameters updated with LARS, because LARS normalizes the gradient. However, batchnorm and bias parameters are excluded from LARS, following SimCLR's and BYOL's protocol, and therefore their gradient is sensitive to --scale-loss. We are currently attempting to remove this --scale-loss by folding it in a specific learning rate for LARS-excluded parameters, but for reasons that we cannot currently explain, this mathematically equivalent formulation does not yield the same accuracy in practice. We hope to find a satisfactory solution soon.

**Comment by mitming** [comment]
If other optimizers are used (e.g. Adam, SGD), will this additional parameter affect the gradient?

**Comment by StphTphsn** [comment]
With Adam or SGD, tuning the learning rate should be equivalent to using the extra hyperparameter --scale-loss.

**Comment by mitming** [comment]
I got it, thanks a lot!
